src/linter/utils.go

```diff
@@ -90,6 +90,32 @@ func typesMapToTypeExpr(p *phpdoc.TypeParser, m meta.TypesMap) phpdoc.Type {
 	return p.Parse(typeString)
 }
 
+// functionReturnType returns the return type of a function over computed types
+// according to the convention below:
+//
+// The types are inferred as follows:
+// 1. If there is an @param annotation, then its value becomes the return type;
```

Comment:
Maybe `@return` annotation was implied here (and below)?

Reply:
Yep. Done.